From: Martin T. H. Sandsmark <martin.sandsmark@kde.org>
Date: Tue, 07 Aug 2012 11:23:28 +0000
Subject: Check if attention demanding status has changed when we activate a window.
X-Git-Url: http://quickgit.kde.org/?p=kde-workspace.git&amp;a=commitdiff&amp;h=d3b3cd48f7b32ef891a85e5321b3c61469ae1df3
---
Check if attention demanding status has changed when we activate a window.

BUG: 303208
---


--- a/libs/taskmanager/task.cpp
+++ b/libs/taskmanager/task.cpp
@@ -168,7 +168,16 @@
 void Task::setActive(bool a)
 {
     d->active = a;
-    emit changed(StateChanged);
+
+    TaskChanges changes = StateChanged;
+
+    if (demandsAttention() != d->demandedAttention) {
+        d->demandedAttention = !d->demandedAttention;
+        changes |= AttentionChanged;
+    }
+
+    emit changed(changes);
+
     if (a) {
         emit activated();
     } else {

